include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef VIGRA_NO_INT_ISNAN
```

Comment:
Maybe this condition can be dropped altogether? The template will only match if there is no explicit implementation of `isnan` for a given type, and then it will do the right thing.